File: spring-cloud-bus-demo/bus-common/src/main/java/com/livk/bus/listener/LivkEventListener.java
package com.livk.bus.listener;

import com.livk.bus.event.LivkBusEvent;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.ApplicationListener;
import org.springframework.lang.Nullable;

/**
 * <p>
 * LivkEventListener
 * </p>
 *
 * @author livk
 * @date 2021/11/1
 */
@Slf4j
public class LivkEventListener implements ApplicationListener<LivkBusEvent> {

	@Override
	public void onApplicationEvent(@Nullable LivkBusEvent event) {
		log.info("listener:{}", event);
	}

}

File: spring-cloud-bus/bus-commons/src/main/kotlin/com/livk/bus/listener/LivkEventListener.kt
package com.livk.bus.listener

import com.livk.bus.event.LivkBusEvent
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.context.ApplicationListener

/**
 * @author livk
 */
class LivkEventListener : ApplicationListener<LivkBusEvent> {
    private val log: Logger = LoggerFactory.getLogger(LivkEventListener::class.java)

    override fun onApplicationEvent(event: LivkBusEvent) {
        log.info("listener:{}", event)
    }
}
